docs: Aliases overview and domain model

Dan-Heath opened the pull request.

AdamBouhmad commented on website/content/docs/concepts/aliases.mdx:
```diff
@@ -0,0 +1,269 @@
+# Aliases
+
+An alias is a resource that lets you associate a string with a specific target.
+The string must be a globally unique value which follows DNS naming constraints.
```
Not necessarily true, as we support hostnames(e.g an alias of `webserver` as well as DNS-like strings `webserver.boundary`. May be worth rephrasing to something like: An alias is a globally unique, DNS-like string associated with a destination resource. Examples of valid aliases may be `webserver` or `webserver.boundary`.

AdamBouhmad commented on website/content/docs/concepts/domain-model/aliases.mdx:
```diff
@@ -0,0 +1,35 @@
+
+An alias is a resource that lets you associate a string with a specific target.
+The string must be a globally unique value which follows DNS naming constraints.
+You can use the alias instead of the target ID when you run a command, and Boundary automatically substitutes the target resource that the alias references.
```
Would maybe simplify this a bit: 

`With aliases, you can establish a session to a target by referencing its alias, instead of having to provide a target-id or target-name & scope id. Example: boundary connect ssh boundary.dev`

AdamBouhmad commented on website/content/docs/concepts/domain-model/aliases.mdx:
```diff
@@ -0,0 +1,35 @@
+
+## Attributes
+
+All alias types have the following attributes:
```
Since there's just one type for the time being, is it worth just saying: 

Aliases have the following attributes:

AdamBouhmad commented on website/content/docs/commands/aliases/list.mdx:
```diff
@@ -0,0 +1,45 @@
+
+Command: `aliases list`
+
+The `aiases list` command lets you list the aliases within a given scope or resource.
```
```suggestion
The `aliases list` command lets you list the aliases within a given scope or resource.
```

AdamBouhmad commented on website/content/docs/commands/delete.mdx:
```diff
@@ -16,6 +16,12 @@ When you run this command, Boundary interprets it as the type-specific command.
 
+<Note>
+
+The generic `boundary delete` command does not work for alias resources.
```
This is supported. 

E.g `boundary delete alt_U1ZksaRyb0`

AdamBouhmad commented on website/content/docs/commands/index.mdx:
```diff
@@ -234,6 +234,10 @@ $ boundary dev -help
+- `-ferry-port=<uint>` - Indicates the port on which the ferry daemon should listen.
+The default port is 9300.
+You can also specify the port using the BOUNDARY_FERRY_LISTENING_PORT environment variable.
+
```
References to ferry should be removed -- not releasing as part of 0.16

AdamBouhmad commented on website/content/docs/commands/index.mdx:
```diff
@@ -255,6 +259,14 @@ client options. Some notable options include the following:
+The default is `false`.
+You can also specify this value using the BOUNDARY_SKIP_CACHE_DAEMON environment variable.
+
+- `-skip-ferry-daemon` - If set, skips sending the auth token used for the command to the ferry daemon, if it is running.
```
References to ferry should be removed -- not releasing as part of 0.16

AdamBouhmad commented on website/content/docs/commands/index.mdx:
```diff
@@ -318,6 +330,22 @@ to the Boundary controller. If you set the path, you must also specify a client
 The path on the local disk to a single PEM-encoded private key that matches the client certificate
 specified by the `BOUNDARY_CLIENT_CERT` variable or the `-client-cert` connection option.
 
+### `BOUNDARY_FERRY_LISTENING_PORT`
```
References to ferry should be removed -- not releasing as part of 0.16

AdamBouhmad commented on website/content/docs/commands/index.mdx:
```diff
@@ -318,6 +330,22 @@ to the Boundary controller. If you set the path, you must also specify a client
+Prevents Boundary from starting the caching demon or sending the current used or retrieved token to the caching daemon.
+You can use the `BOUNDARY_SKIP_CACHE_DAEMON` variable or the `-skip-cache-daemon` connection option.
+
+### `BOUNDARY_SKIP_FERRY_DAEMON`
```
References to ferry should be removed -- not releasing as part of 0.16

AdamBouhmad commented on website/content/docs/commands/read.mdx:
```diff
@@ -16,6 +16,12 @@ When you run this command, Boundary interprets it as the type-specific command.
 
+<Note>
+
+The generic `boundary read` command does not work for alias resources.
```
this does work, e.g: 
```
bouhmad :: boundary read alt_8H2vEcQenC
Alias information:
  Created Time:        Thu, 25 Apr 2024 11:58:24 PDT
  Destination ID:      ttcp_uWQ4GJptNC
  ID:                  alt_8H2vEcQenC
  Type:                target
  Updated Time:        Thu, 25 Apr 2024 11:58:24 PDT
  Value:               www.hashicorp.com
  Version:             1

  Scope:
    ID:                global
    Name:              global
    Type:              global

  Authorized Actions:
    update
    delete
    no-op
    read

AdamBouhmad commented on website/content/docs/commands/targets/create.mdx:
```diff
@@ -115,6 +121,12 @@ If you do not specify a default port, Boundary uses port 22.
 - `-session-max-seconds=<string>` - The maximum lifetime of the session, including all connections.
 You can specify an integer number of seconds or a duration string.
+- `-with-alias-authorize-session-host-id=<string>` - The host ID that an alias uses to authorize sessions for the target.
+- `-with-aliasscope-id=<string>` - The scope ID that you want to create the target and alias in.
```
```suggestion
- `-with-alias-scope-id=<string>` - The scope ID that you want to create the target and alias in.
```

AdamBouhmad commented on website/content/docs/commands/update.mdx:
```diff
@@ -16,6 +16,12 @@ When you run this command, Boundary interprets it as the type-specific command.
 
+<Note>
+
+The generic `boundary update` command does not work for alias resources.
```
This is supported

AdamBouhmad commented on website/content/docs/concepts/aliases.mdx:
```diff
@@ -0,0 +1,269 @@
+When you create a resource in Boundary, it assigns the resource a randomly generated identifier.
+You must use those IDs to perform actions in Boundary.
+
+For example, when you connect to a target using the clipboard, you must reference the target ID by typing `boundary ssh connect -target-id tccp_123456789`.
```
``
```suggestion
When you connect to a target using the terminal, you must reference the target ID or target-name & scope-name. As an example, to SSH to a target, you can type `boundary connect ssh -target-id ttcp_123456789`.
```

AdamBouhmad commented on website/content/docs/concepts/aliases.mdx:
```diff
@@ -0,0 +1,269 @@
+   - **Type**: Select **Target**.
+   At this time, targets are the only Boundary resources that supports aliasing.
+   - **Alias Value**: Enter the alias value you want to use in commands to represent the target.
+   It must comply with DNS naming rules.
```
```suggestion
   An alias's value can be a hostname or a DNS-like string.
```

AdamBouhmad commented on website/content/docs/concepts/aliases.mdx:
```diff
@@ -0,0 +1,269 @@
+   For unlimited connections, enter `-1`.
+   - **Workers**: (Optional) Select whether you want the worker to function as an ingress and/or egress worker.
+   - **Aliases**: (Optional) Enter the value fpr any aliases you want to associate with this target, and then click **Add**.
+   The value must match DNS naming rules.
```
```suggestion
   An alias's value can be a hostname or a DNS-like string.
```

AdamBouhmad commented on website/content/docs/concepts/aliases.mdx:
```diff
@@ -0,0 +1,269 @@
+   - **Type**: Select **Target**.
+   At this time, targets are the only Boundary resources that supports aliasing.
+   - **Alias Value**: Enter the string that you want to use as the alias to represent the target.
+   The alias value must comply with DNS naming rules.
```
```suggestion
   An alias's value can be a hostname or a DNS-like string.
```

AdamBouhmad commented on website/content/docs/concepts/aliases.mdx:
```diff
@@ -0,0 +1,269 @@
+   You can also specify the scope using the BOUNDARY_SCOPE_ID environment variable.
+   At this time, aliases are only supported for the global scope.
+   - `-value=<string>` - Specifies the string that you want to use as the alias to represent the target.
+   The alias `value` must comply with DNS naming rules.
```
```suggestion
   The alias `value` can be a hostname or a DNS-like string.
```

AdamBouhmad commented on website/content/docs/concepts/aliases.mdx:
```diff
@@ -0,0 +1,269 @@
+- Cannot start or end with a hyphen ( - )
+- Between 1 and 63 characters long
+- The final label cannot be all numeric characters
+
```
We should note that it does not have to be delimited by a suffix, and can just be a hostname. 

E.g the value can just be `webserver` instead of `webserver.boundary`

AdamBouhmad reviewed (changes requested):
Fantastic start and extremely thorough, awesome job @Dan-Heath. I left a few comments for things I think should be modified, but otherwise, this looks great

Dan-Heath commented:
Thanks for the reviews and feedback, @AdamBouhmad and @talanknight ! I accepted all of your suggestions. I deleted the incorrect notes about the generic commands not working with aliases, deleted references to ferry, and revised the naming conventions. I think this addresses all the feedback. Please take a look and let me know what you think. Thanks again!

stellarsquall commented on website/content/docs/concepts/domain-model/aliases.mdx:
```diff
@@ -0,0 +1,36 @@
+An alias is a globally unique, DNS-like string that is associated wtih a destination resource.
+Examples of valid aliases are `webserver` and `webserver.boundary`.
+You can establish a session to a target by referencing its alias, instead of having to provide a target ID or target name and scope ID.
+For example, if you have an alias `boundary.dev`, you can use it to connect to a target with the following command: `boundary connect ssh boundary.dev`.
```
I'd prefer this to be shown using shell format, like:

> For example, if you have an alias `boundary.dev`, you can use it to connect to a target with the following command: 

```shell-session
$ boundary connect ssh boundary.dev
```

stellarsquall commented on website/content/docs/concepts/aliases.mdx:
```diff
@@ -0,0 +1,264 @@
+An alias is a globally unique, DNS-like string that is associated with a destination resource.
+Exampes of valid aliases include `webserver` and `webserver.boundary`.
+You can establish a session to a target by referencing its alias, instead of having to provide a target ID or target name and scope ID.
+For example, if you have an alias `boundary.dev`, you can use it to connect to a target with the following command: `boundary connect ssh boundary.dev`.
```
Same here, would prefer to see this broken out into shell-session format

stellarsquall commented on website/content/docs/concepts/aliases.mdx:
```diff
@@ -0,0 +1,264 @@
+When you create a resource in Boundary, it assigns the resource a randomly generated identifier.
+You must use those IDs to perform actions in Boundary.
+
+When you connect to a target using the terminal, you must reference the target ID or target name and scope name. As an example, to SSH to a target, you can type `boundary connect ssh -target-id ttcp_123456789`.
```
```suggestion
When you connect to a target using the terminal, you must reference the target ID or target name and scope name. As an example, to SSH to a target, you can execute the command `boundary connect ssh -target-id ttcp_123456789`.
```

stellarsquall commented on website/content/docs/concepts/aliases.mdx:
```diff
@@ -0,0 +1,264 @@
+
+An alias is a globally unique, DNS-like string that is associated with a destination resource.
+The alias `value` parameter does not have to be delimited by a suffix, and can be just a hostname.
+Exampes of valid aliases include `webserver` and `webserver.boundary`.
```
This sentence was repeated at the top of the page. Are there other valid examples or naming formats we want to call out here? Or are these only valid formats:

`value1`
`value1.value2.valueX`

stellarsquall commented on website/content/docs/concepts/aliases.mdx:
```diff
@@ -0,0 +1,264 @@
+## Scopes
+
+At this time, you can only create aliases in the `global` scope.
+An alias' destination target can be in any scope, however.
```
I'd suggest removing this possessive for accessibility 

```suggestion
The destination target for an alias can be in any scope, however.
```

stellarsquall commented on website/content/docs/concepts/aliases.mdx:
````diff
@@ -0,0 +1,264 @@
+For example, you can use the following command to connect to an SSH target with the ID `ttcp_1234567890`:
+
+```shell-session
+boundary connect ssh -target -id ttcp_1234567890
````
```suggestion
$ boundary connect ssh -target -id ttcp_1234567890
```

stellarsquall commented on website/content/docs/concepts/aliases.mdx:
````diff
@@ -0,0 +1,264 @@
+If you configured an alias named `example.alias.boundary` for the target, you could alternatively use the following command to connect to the target:
+
+```shell-session
+boundary connect ssh example.alias.boundary
````
```suggestion
$ boundary connect ssh example.alias.boundary
```

stellarsquall commented on website/content/docs/concepts/aliases.mdx:
```diff
@@ -0,0 +1,264 @@
+
+</Tabs>
+
+### Create a new target and alias
```
I found this header a little confusing, just suggesting different verbage

```suggestion
### Create an alias during target creation
```

Dan-Heath replied:
I like this much better. Thanks!

stellarsquall commented on website/content/docs/concepts/aliases.mdx:
```diff
@@ -0,0 +1,264 @@
+- [Create a new target and alias at the same time](/boundary/docs/concepts/aliases#create-a-new-target-and-alias)
+- [Associate an existing alias with a target](/boundary/docs/concepts/aliases#associate-an-existing-alias-with-a-target)
+
+### Create a new alias for an existing target
```
this feels more natural to me

```suggestion
### Create an alias for an existing target
```
